### DISCUSSION TITLE
Hello everyone, please help me I'm stupid

### DISCUSSION THREAD

**Discussion by Mikuo39** [lshqqytiger/stable-diffusion-webui-directml#356] [Q&A] [answered]
i have rx580 8gb and fx8320 ,  https://www.youtube.com/watch?v=km5Jy3w9soE  I followed the instructions in this video and also did everything according to the guide from https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Install-and-Run-on-AMD-GPUs 
but nothing helps me, I end up getting the error [notice] A new release of pip available: 22.2.1 -> 23.3.2
[notice] To update, run: D:\webui\stable-diffusion-webui-directml\venv\Scripts\python.exe -m pip install --upgrade pip
Traceback (most recent call last):
   File "D:\webui\stable-diffusion-webui-directml\launch.py", line 48, in <module>
     main()
   File "D:\webui\stable-diffusion-webui-directml\launch.py", line 39, in main
     prepare_environment()
   File "D:\webui\stable-diffusion-webui-directml\modules\launch_utils.py", line 560, in prepare_environment
     raise RuntimeError(
RuntimeError: Torch is not able to use GPU; add --skip-torch-cuda-test to COMMANDLINE_ARGS variable to disable this check
To continue, press any key. . . 
----------------
my cmd 
Microsoft Windows [Version 10.0.19045.3448]
(c) Корпорация Майкрософт (Microsoft Corporation). Все права защищены.

D:\webui>git clone https://github.com/lshqqytiger/stable-diffusion-webui-directml && cd stable-diffusion-webui-directml && git submodule init && git submodule update
Cloning into 'stable-diffusion-webui-directml'...
remote: Enumerating objects: 30292, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 30292 (delta 0), reused 2 (delta 0), pack-reused 30289
Receiving objects: 100% (30292/30292), 33.83 MiB | 5.38 MiB/s, done.
Resolving deltas: 100% (21150/21150), done.

D:\webui\stable-diffusion-webui-directml>
--------------------------------------------------------------------------
all text  in webui-user.bat
Creating venv in directory D:\webui\stable-diffusion-webui-directml\venv using python "C:\Users\Ivy\AppData\Local\Programs\Python\python.exe"
venv "D:\webui\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: 1.7.0
Commit hash: b6ebcbf458c498741ac34bd67328edd58c1dc108
Installing torch and torchvision
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cu118
Collecting torch==2.0.1
  Using cached https://download.pytorch.org/whl/cu118/torch-2.0.1%2Bcu118-cp310-cp310-win_amd64.whl (2619.1 MB)
Collecting torchvision==0.15.2
  Using cached https://download.pytorch.org/whl/cu118/torchvision-0.15.2%2Bcu118-cp310-cp310-win_amd64.whl (4.9 MB)
Collecting typing-extensions
  Using cached typing_extensions-4.9.0-py3-none-any.whl (32 kB)
Collecting networkx
  Using cached networkx-3.2.1-py3-none-any.whl (1.6 MB)
Collecting jinja2
  Using cached Jinja2-3.1.3-py3-none-any.whl (133 kB)
Collecting filelock
  Using cached filelock-3.13.1-py3-none-any.whl (11 kB)
Collecting sympy
  Using cached https://download.pytorch.org/whl/sympy-1.12-py3-none-any.whl (5.7 MB)
Collecting numpy
  Using cached numpy-1.26.3-cp310-cp310-win_amd64.whl (15.8 MB)
Collecting pillow!=8.3.*,>=5.3.0
  Using cached pillow-10.2.0-cp310-cp310-win_amd64.whl (2.6 MB)
Collecting requests
  Using cached requests-2.31.0-py3-none-any.whl (62 kB)
Collecting MarkupSafe>=2.0
  Using cached https://download.pytorch.org/whl/MarkupSafe-2.1.3-cp310-cp310-win_amd64.whl (17 kB)
Collecting charset-normalizer<4,>=2
  Using cached charset_normalizer-3.3.2-cp310-cp310-win_amd64.whl (100 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2023.11.17-py3-none-any.whl (162 kB)
Collecting urllib3<3,>=1.21.1
  Using cached urllib3-2.1.0-py3-none-any.whl (104 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.6-py3-none-any.whl (61 kB)
Collecting mpmath>=0.19
  Using cached https://download.pytorch.org/whl/mpmath-1.3.0-py3-none-any.whl (536 kB)
Installing collected packages: mpmath, urllib3, typing-extensions, sympy, pillow, numpy, networkx, MarkupSafe, idna, filelock, charset-normalizer, certifi, requests, jinja2, torch, torchvision
Successfully installed MarkupSafe-2.1.3 certifi-2023.11.17 charset-normalizer-3.3.2 filelock-3.13.1 idna-3.6 jinja2-3.1.3 mpmath-1.3.0 networkx-3.2.1 numpy-1.26.3 pillow-10.2.0 requests-2.31.0 sympy-1.12 torch-2.0.1+cu118 torchvision-0.15.2+cu118 typing-extensions-4.9.0 urllib3-2.1.0

[notice] A new release of pip available: 22.2.1 -> 23.3.2
[notice] To update, run: D:\webui\stable-diffusion-webui-directml\venv\Scripts\python.exe -m pip install --upgrade pip
Traceback (most recent call last):
  File "D:\webui\stable-diffusion-webui-directml\launch.py", line 48, in <module>
    main()
  File "D:\webui\stable-diffusion-webui-directml\launch.py", line 39, in main
    prepare_environment()
  File "D:\webui\stable-diffusion-webui-directml\modules\launch_utils.py", line 560, in prepare_environment
    raise RuntimeError(
RuntimeError: Torch is not able to use GPU; add --skip-torch-cuda-test to COMMANDLINE_ARGS variable to disable this check
Для продолжения нажмите любую клавишу . . .

**Comment by lshqqytiger** [comment]
Add `--use-directml`.

**Reply by Mikuo39** [reply]
Hello, sorry my questions are not very smart, where exactly should I enter this? in webui-user.bat where set COMMANDLINE_ARGS=

**Reply by Mikuo39** [reply]
Thank you very much for trying to help, after I entered set COMMANDLINE_ARGS=--use-directml
then the following appeared 
venv "D:\sd\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: 1.7.0
Commit hash: b6ebcbf458c498741ac34bd67328edd58c1dc108
Installing clip
Installing open_clip
Cloning Stable Diffusion into D:\sd\stable-diffusion-webui-directml\repositories\stable-diffusion-stability-ai...
Cloning into 'D:\sd\stable-diffusion-webui-directml\repositories\stable-diffusion-stability-ai'...
remote: Enumerating objects: 580, done.
remote: Counting objects: 100% (310/310), done.
remote: Compressing objects: 100% (92/92), done.
remote: Total 580 (delta 248), reused 218 (delta 218), pack-reused 270Receiving objects:  93% (540/580), 68.74 MiB | 10.77 MiB/s
Receiving objects: 100% (580/580), 73.43 MiB | 10.42 MiB/s, done.
Resolving deltas: 100% (280/280), done.
Cloning Stable Diffusion XL into D:\sd\stable-diffusion-webui-directml\repositories\generative-models...
Cloning into 'D:\sd\stable-diffusion-webui-directml\repositories\generative-models'...
remote: Enumerating objects: 860, done.
remote: Counting objects: 100% (503/503), done.
remote: Compressing objects: 100% (233/233), done.
remote: Total 860 (delta 359), reused 309 (delta 266), pack-reused 357
Receiving objects: 100% (860/860), 42.67 MiB | 9.66 MiB/s, done.
Resolving deltas: 100% (435/435), done.
Cloning K-diffusion into D:\sd\stable-diffusion-webui-directml\repositories\k-diffusion...
Cloning into 'D:\sd\stable-diffusion-webui-directml\repositories\k-diffusion'...
remote: Enumerating objects: 1329, done.
remote: Counting objects: 100% (611/611), done.
remote: Compressing objects: 100% (81/81), done.
remote: Total 1329 (delta 568), reused 538 (delta 530), pack-reused 718
Receiving objects: 100% (1329/1329), 239.04 KiB | 855.00 KiB/s, done.
Resolving deltas: 100% (931/931), done.
Cloning CodeFormer into D:\sd\stable-diffusion-webui-directml\repositories\CodeFormer...
Cloning into 'D:\sd\stable-diffusion-webui-directml\repositories\CodeFormer'...
remote: Enumerating objects: 594, done.
remote: Counting objects: 100% (245/245), done.
remote: Compressing objects: 100% (90/90), done.
remote: Total 594 (delta 176), reused 170 (delta 155), pack-reused 349
Receiving objects: 100% (594/594), 17.30 MiB | 4.46 MiB/s, done.
Resolving deltas: 100% (287/287), done.
Cloning BLIP into D:\sd\stable-diffusion-webui-directml\repositories\BLIP...
Cloning into 'D:\sd\stable-diffusion-webui-directml\repositories\BLIP'...
remote: Enumerating objects: 277, done.
remote: Counting objects: 100% (165/165), done.
remote: Compressing objects: 100% (30/30), done.
remote: Total 277 (delta 137), reused 136 (delta 135), pack-reused 112
 MiB/s
Receiving objects: 100% (277/277), 7.03 MiB | 5.44 MiB/s, done.
Resolving deltas: 100% (152/152), done.
Installing requirements for CodeFormer
Installing requirements
Launching Web UI with arguments: --use-directml
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
Traceback (most recent call last):
  File "D:\sd\stable-diffusion-webui-directml\launch.py", line 48, in <module>
    main()
  File "D:\sd\stable-diffusion-webui-directml\launch.py", line 44, in main
    start()
  File "D:\sd\stable-diffusion-webui-directml\modules\launch_utils.py", line 687, in start
    import webui
  File "D:\sd\stable-diffusion-webui-directml\webui.py", line 13, in <module>
    initialize.imports()
  File "D:\sd\stable-diffusion-webui-directml\modules\initialize.py", line 34, in imports
    shared_init.initialize()
  File "D:\sd\stable-diffusion-webui-directml\modules\shared_init.py", line 26, in initialize
    dml.do_hijack()
  File "D:\sd\stable-diffusion-webui-directml\modules\dml\__init__.py", line 74, in do_hijack
    if not torch.dml.has_float64_support(device):
AttributeError: module 'torch' has no attribute 'dml'
Для продолжения нажмите любую клавишу . . .

**Reply by lshqqytiger** [reply]
You may not have torch-directml installed. If restarting does not work, add new line `torch-directml` in requirements.txt and try again.

**Reply by Mikuo39** [reply]
Sorry, but it didn't help, thanks for wasting your time on me 
venv "D:\sd\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: 1.7.0
Commit hash: b6ebcbf458c498741ac34bd67328edd58c1dc108
Launching Web UI with arguments: --use-directml
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
Traceback (most recent call last):
  File "D:\sd\stable-diffusion-webui-directml\launch.py", line 48, in <module>
    main()
  File "D:\sd\stable-diffusion-webui-directml\launch.py", line 44, in main
    start()
  File "D:\sd\stable-diffusion-webui-directml\modules\launch_utils.py", line 687, in start
    import webui
  File "D:\sd\stable-diffusion-webui-directml\webui.py", line 13, in <module>
    initialize.imports()
  File "D:\sd\stable-diffusion-webui-directml\modules\initialize.py", line 34, in imports
    shared_init.initialize()
  File "D:\sd\stable-diffusion-webui-directml\modules\shared_init.py", line 26, in initialize
    dml.do_hijack()
  File "D:\sd\stable-diffusion-webui-directml\modules\dml\__init__.py", line 74, in do_hijack
    if not torch.dml.has_float64_support(device):
AttributeError: module 'torch' has no attribute 'dml'
Для продолжения нажмите любую клавишу . . .

**Reply by Mikuo39** [reply]
my requirements.txt 
GitPython
Pillow
accelerate

basicsr
blendmodes
clean-fid
diffusers
einops
fastapi>=0.90.1
gfpgan
gradio==3.41.2
inflection
jsonmerge
kornia
lark
numpy
omegaconf
open-clip-torch

piexif
psutil
pytorch_lightning
realesrgan
requests
resize-right

safetensors
scikit-image>=0.19
timm
tomesd
torch>=2.0.0
torch-directml
torchdiffeq
torchsde
transformers==4.30.2

**Reply by lshqqytiger** [reply, marked as answer]
Oh, my bad. Instead of `requirements.txt`, it was `requirements_version.txt`.

**Reply by Mikuo39** [reply]
Thank you very much for the time you spent on me and the work you did, I did as you wrote and everything worked.

**Comment by Mikuo39** [comment]
Now I watched this video and did everything from start to finish, in the end I get the same error again https://youtu.be/nd4Q60lLWbg
venv "D:\sd\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: 1.7.0
Commit hash: b6ebcbf458c498741ac34bd67328edd58c1dc108
Traceback (most recent call last):
  File "D:\sd\stable-diffusion-webui-directml\launch.py", line 48, in <module>
    main()
  File "D:\sd\stable-diffusion-webui-directml\launch.py", line 39, in main
    prepare_environment()
  File "D:\sd\stable-diffusion-webui-directml\modules\launch_utils.py", line 560, in prepare_environment
    raise RuntimeError(
RuntimeError: Torch is not able to use GPU; add --skip-torch-cuda-test to COMMANDLINE_ARGS variable to disable this check
Для продолжения нажмите любую клавишу . . .
